### PR TITLE
Api Call Refactor

### PR DESCRIPTION
**Pull request by borisyankov**
Consolidate api call logic and remove code duplication

### CODE REVIEW

File: src/api/apiFetch.js [hunk 1]
```diff
@@ -8,31 +9,84 @@ export type Auth = {
...
 const apiVersion = 'api/v1';
 
-export const getAuthHeader = (email, apiKey) => {
-  const encodedStr = `${email}:${apiKey}`;
-  return `Basic ${base64.encode(encodedStr)}`;
-};
+export const getAuthHeader = (email, apiKey) =>
+  `Basic ${base64.encode(`${email}:${apiKey}`)}`;
+
+// fetch("http://httpstat.us/500")
```
Comment (neerajwahi) [inline src/api/apiFetch.js]: Please delete this comment

File: src/api/apiFetch.js [hunk 2]
```diff
@@ -8,31 +9,84 @@ export type Auth = {
...
+
 
-export const apiFetch = async (authObj: Auth, route: string, params: Object) => {
+export const apiFetch = async (
+  authObj: Auth,
+  route: string,
+  params: Object,
+  doNotTimeout: boolean = false,
```
Comment (neerajwahi) [inline src/api/apiFetch.js]: I'd probably name this `shouldTimeout` (seems clearer to me) but it's up to you

File: src/api/apiFetch.js [hunk 3]
```diff
@@ -8,31 +9,84 @@ export type Auth = {
...
+  let timeout;
   try {
-    const res = await raw.json();
-    return res;
-  } catch (err) {
-    throw new Error(`HTTP response code ${raw.status}: ${raw.statusText}`);
+    if (!doNotTimeout) {
+      timeout = setTimeout(() => { throw Error(`Request timed out @ ${route}`); }, 5000);
+    }
+    console.log('setTimeout', timeout);
```
Comment (neerajwahi) [inline src/api/apiFetch.js]: Please remove the `console.log`

File: src/api/apiFetch.js [hunk 4]
```diff
@@ -8,31 +9,84 @@ export type Auth = {
...
+    const response = await fetch(url, allParams);
+
+    // console.log('ERROR', err); redux => action for error
```
Comment (neerajwahi) [inline src/api/apiFetch.js]: Please delete unused comments

File: src/api/apiFetch.js [hunk 5]
```diff
@@ -8,31 +9,84 @@ export type Auth = {
...
+    if (!response.ok) throw Error(response.statusText);
+    // Error(`HTTP response code ${response.status}: ${response.statusText}`);
+
+    return await response.json();
+  } finally {
+    console.log('clearTimeout', timeout);
```
Comment (neerajwahi) [inline src/api/apiFetch.js]: Ditto

File: src/api/apiFetch.js [hunk 6]
```diff
@@ -8,31 +9,84 @@ export type Auth = {
...
+    clearTimeout(timeout);
+  }
+};
+
+export const apiGet = async (
```
Comment (neerajwahi) [inline src/api/apiFetch.js]: Why have separate functions for GET and POST? Seems like a lot of code duplication

**Comment by borisyankov** [comment]
I have taken into account your feedback and have a commit or two for tomorrow.
If you want you can merge this one though.

File: src/app/appActions.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+export const APP_ACTIVITY = 'APP_ACTIVITY';
+export const APP_ONLINE = 'APP_OFFLINE';
+export const APP_OFFLINE = 'APP_OFFLINE';
+
+export const appActivity = (auth) =>
+  async (dispatch) =>
+    await focusPing(auth, true, false);
+
+export const appOniline = () => ({
```
Comment (neerajwahi) [inline src/app/appActions.js]: Typo: should be `appOnline`

Comment (borisyankov) [inline src/app/appActions.js]: I did fix this in the next commit that I haven't pushed.

**Comment by neerajwahi** [comment]
@borisyankov The app seems broken in the simulator when I test this PR. I get an error when fetching streams

**Comment by borisyankov** [comment]
Something screwed up with merge/rebase I guess.
I can reproduce. Will fix.

**Comment by borisyankov** [comment]
Closing this. Fixed in the 'actions' PR that is based off of this.

**Pull request closed without merging**